external/llvm-project/mlir/lib/Dialect/AMDGPU/IR/AMDGPUDialect.cpp

```diff
@@ -326,6 +326,36 @@ LogicalResult MFMAOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// DPPOp
+//===----------------------------------------------------------------------===//
+LogicalResult DPPOp::verify() {
+  Type srcType = getSrc().getType();
+  if (srcType.getIntOrFloatBitWidth() > 32) {
+    return emitOpError("integer and floating point types larger than 32 bits "
+                       "are not supported");
+  }
+
+  llvm::ArrayRef<std::string> attributeNames = {
```

Comment:
The core reason I want an enum for this is because someone who's just skimming the tablegen file will not understand that they're only allowed to pick one.

Reply:
I understand your point, but I haven't seen any example to model an sum type that is not an plain C-like enum. Do you perhaps have one? The best way I could think of, as you see, is to enforce this constraint in validation.